### PR TITLE
Backoff and retry reindexed finished write

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This commit modifies the ReindexTaskStateUpdater to backoff when a
finished write to the .reindex index fails. If the write is due to a
version conflict exception, it will not attempt to proceed similar to
the checkpoint writes.

### CODE REVIEW

File: modules/reindex/src/main/java/org/elasticsearch/index/reindex/ReindexTaskStateUpdater.java [hunk 1]
```diff
@@ -154,28 +155,42 @@ public void onFailure(Exception e) {
...
             }
         });
     }
+
+    private TimeValue getNextDelay(TimeValue delay) {
+        TimeValue newDelay;
+        if (TimeValue.ZERO.equals(delay)) {
+            newDelay = TimeValue.timeValueMillis(500);
+        } else {
+            newDelay = TimeValue.timeValueSeconds(Math.max(delay.getMillis() * 2, TimeValue.timeValueMinutes(1).millis()));
```
Comment (henningandersen) [inline modules/reindex/src/main/java/org/elasticsearch/index/reindex/ReindexTaskStateUpdater.java]: `timeValueSeconds` looks like it should be `timeValueMillis`?

I think we should raise the max to something like 30 minutes.

As a follow-up, we should consider observing the cluster state and retrigger whenever a shard of the index goes from RED to YELLOW (or something similar)?

**Review by henningandersen — APPROVED** [review]
LGTM

**Comment by tbrooks8** [comment]
@elasticmachine run elasticsearch-ci/2
@elasticmachine run elasticsearch-ci/packaging-sample

**Comment by tbrooks8** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Comment by tbrooks8** [comment]
@elasticmachine run elasticsearch-ci/2

**Pull request merged**